ci/jenkins/jobs/macros.yaml

```diff
@@ -127,3 +127,16 @@
           fi
           echo "=== FAILURE !!! ==="
           exit 1
+
+- builder:
+    name: builder-matrix
+    builders:
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          export JOB_NAME="matrix-${TEST_OS}-k8s-${K8S_VERSION//./-}-build-num"
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --setup-only
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --testcase e2e --test-only
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --testcase 'networkpolicy' --test-only
+          ./ci/jenkins/test-vmc.sh --cluster-name "${JOB_NAME}-${BUILD_NUMBER}" --testcase 'whole-conformance' --test-only
```

Comment:
If any of the test fails, the script will abort immediately because we set "set -e", the cleanup step in line 142 will not be run.

I think maybe we can do something like:

```
e2e_fail=0
np__fail=0
conformance_fail=0
./ci/jenkins/test-vmc.sh ... e2e ... || e2e_fail=1
./ci/jenkins/test-vmc.sh ... networkpolicy ... || np_fail=1
./ci/jenkins/test-vmc.sh ... whole-conformance... || conformance_fail=1
./ci/jenkins/test-vmc.sh ... --cleanup-only
if [ "${e2e_fail}" -eq 1 ]; then echo "E2E test failed!" ; fi
... # same for np_fail and conformance_fail
exit $((e2e_fail + np_fail + conformance_fail))
```
```

Reply:
Script added.